packages/react-component-library/src/components/TextInput/TextInput.tsx

```diff
@@ -98,6 +98,14 @@ const Component = React.forwardRef<HTMLInputElement, TextInputProps>(
   ) => {
     const { hasFocus, onLocalBlur, onLocalFocus } = useFocus(onBlur)
 
+    const inputRef = useRef<HTMLInputElement>(null)
+
+    useImperativeHandle(ref, () => inputRef.current!)
```

Comment:
Another technique used elsewhere is to use mergeRefs:

https://github.com/Royal-Navy/design-system/blob/eb9b42d198c8ac604528dd2b1139cfb2e62f90c6/packages/react-component-library/src/components/CheckboxRadioBase/CheckboxRadioBase.tsx#L128

This way we don't need to decorate the optional forwarded ref.